plugin/src/main/java/org/opensearch/ml/action/models/DeleteModelTransportAction.java

```diff
@@ -44,36 +55,70 @@ public class DeleteModelTransportAction extends HandledTransportAction<ActionReq
     static final String SEARCH_FAILURE_MSG = "Search failure while deleting model of ";
     static final String OS_STATUS_EXCEPTION_MESSAGE = "Failed to delete all model chunks";
     Client client;
+    NamedXContentRegistry xContentRegistry;
 
     @Inject
-    public DeleteModelTransportAction(TransportService transportService, ActionFilters actionFilters, Client client) {
+    public DeleteModelTransportAction(
+        TransportService transportService,
+        ActionFilters actionFilters,
+        Client client,
+        NamedXContentRegistry xContentRegistry
+    ) {
         super(MLModelDeleteAction.NAME, transportService, actionFilters, MLModelDeleteRequest::new);
         this.client = client;
+        this.xContentRegistry = xContentRegistry;
     }
 
     @Override
     protected void doExecute(Task task, ActionRequest request, ActionListener<DeleteResponse> actionListener) {
         MLModelDeleteRequest mlModelDeleteRequest = MLModelDeleteRequest.fromActionRequest(request);
         String modelId = mlModelDeleteRequest.getModelId();
-
-        DeleteRequest deleteRequest = new DeleteRequest(ML_MODEL_INDEX, modelId);
+        MLModelGetRequest mlModelGetRequest = new MLModelGetRequest(modelId, true);
```

Comment:
Do we need to return model content here?

Reply:
Yeah we can skip it.